packages/embeddable/src/index.tsx

```
@@ -0,0 +1,77 @@
+import React, { lazy, Suspense } from "react";
+import ReactDOM from "react-dom";
+import root from "react-shadow";
+import { CourierProvider } from "@trycourier/react-provider";
+
+const Toast = lazy(() => import("./components/Toast"));
+const Inbox = lazy(() => import("./components/Inbox"));
+
+declare global {
+  interface Window {
+    courierConfig: {
+      clientKey: string;
+      apiUrl?: string;
+      userId?: string;
+      userSignature?: string;
+    };
+  }
+}
+
+const CourierComponents: React.FunctionComponent = () => {
+  const elements = Array.from(document.querySelectorAll("courier"));
+
+  return (
+    <>
+      {elements.map((element) => {
+        const componentType = element.getAttribute("component");
+        const Component = (() => {
+          switch (componentType) {
+            case "toast": {
+              return Toast;
+            }
+
+            case "inbox": {
+              return Inbox;
+            }
+          }
+        })();
+
+        if (!Component) {
+          return null;
+        }
+
+        return ReactDOM.createPortal(
+          <Suspense fallback={<div />}>
+            <Component />
+          </Suspense>,
+          element
+        );
+      })}
+    </>
+  );
+};
+
+(async () => {
+  const { clientKey, apiUrl, userId, userSignature } = window.courierConfig;
```

Should we throw an error if correct props are not found on window?